### PR TITLE
riverctl: Error messages instead of error traces for common user errors

### PR DESCRIPTION
**Pull request by Leon-Plickat**

### CODE REVIEW

**Review by ifreund — COMMENTED** [review]
instead of duplicating the same error message in several places how about we do something like this?
```zig
pub fn main() void {
    real_main() catch |err| switch (err) {
        error.RiverControlNotAdvertised => printErrorExit(...),
        error.Foo ...
        ...
    };
)
```

**Comment by Leon-Plickat** [comment]
So we basically put a function between `_root` and `main`? Seems good.

File: riverctl/main.zig [hunk 1]
```diff
@@ -44,6 +44,30 @@ pub const Globals = struct {
...
+        , .{}),
+        error.RiverOptionsManagerNotAdvertised => printErrorExit(
+            \\The Wayland server does not support river-options-unstable-v1.
+            \\Do your versions of river and riverctl match?
+        , .{}),
+        error.SeatNotAdverstised => printErrorExit(
+            \\The Wayland server does not advertise any seat.
+        , .{}),
+        error.XdgOutputNotAdvertised => printErrorExit(
+            \\The Wayland server does not support the XDG-output.
```
Comment (ifreund) [inline riverctl/main.zig:64]:
```suggestion
            \\The Wayland server does not support xdg-output-unstable-v1.
```

File: riverctl/main.zig [hunk 2]
```diff
@@ -44,6 +44,30 @@ pub const Globals = struct {
...
+        error.RiverStatusManagerNotAdvertised => printErrorExit(
+            \\The Wayland server does not support river-status-unstable-v1.
+            \\Do your versions of river and riverctl match?
+        , .{}),
+        error.RiverOptionsManagerNotAdvertised => printErrorExit(
+            \\The Wayland server does not support river-options-unstable-v1.
+            \\Do your versions of river and riverctl match?
+        , .{}),
+        error.SeatNotAdverstised => printErrorExit(
+            \\The Wayland server does not advertise any seat.
```
Comment (ifreund) [inline riverctl/main.zig:61]:
```suggestion
            \\The Wayland server did not advertise any seat.
```

**Review by ifreund — COMMENTED** [review]
Looks much cleaner :)

**Review by ifreund — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**